src/utilities/toast-manager.test.ts

```diff
@@ -1,92 +1,193 @@
-        // Arrange
-        const testToastId = faker.random.number(99);
-        const toastContent = faker.random.word();
-        const toastOptions = { toastId: testToastId };
+    // #endregion success
+
+    // -----------------------------------------------------------------------------------------
+    // #region warn
+    // -----------------------------------------------------------------------------------------
 
-        // Act
-        const toastId = ToastManager.warn(toastContent, toastOptions);
+    describe("warn", () => {
+        test("when toast created with toastId, returns given toastId", () => {
+            // Arrange
+            const expected = faker.random.number(99);
+            const toastContent = faker.random.word();
+            const toastOptions = { toastId: expected };
 
-        // Assert
-        expect(toastId).toBe(testToastId);
+            // Act
+            const toastId = ToastManager.warn(toastContent, toastOptions);
+
+            // Assert
+            expect(toastId).toBe(expected);
+        });
+
+        test("when toast created without toastId, returns new toastId", () => {
+            // Arrange
+            const toastContent = faker.random.word();
+
+            // Act
+            const toastId = ToastManager.warn(toastContent);
+
+            // Assert
+            expect(toastId).not.toBeNil();
+        });
     });
 
-    test("when toast dismissed, calls dismiss method", () => {
-        // Arrange
-        const dismissMethodSpy = jest.spyOn(ToastManager, "dismiss");
-        const testId = faker.random.number(99);
+    // #endregion warn
+
+    // -----------------------------------------------------------------------------------------
+    // #region dismiss
+    // -----------------------------------------------------------------------------------------
+
+    describe("dismiss", () => {
+        test("when executed, calls dismiss", () => {
+            // Arrange
+            const dismissMethodSpy = jest.spyOn(toast, "dismiss");
+            const testId = faker.random.number(99);
 
-        // Act
-        ToastManager.dismiss(testId);
+            // Act
+            ToastManager.dismiss(testId);
 
-        // Assert
-        expect(dismissMethodSpy).toHaveBeenCalled();
+            // Assert
+            expect(dismissMethodSpy).toHaveBeenCalledWith(testId);
+        });
     });
 
-    test("when dismiss All toasts, calls dismissAll method", () => {
-        // Arrange
-        const dismissAllMethodSpy = jest.spyOn(ToastManager, "dismissAll");
+    // #endregion dismiss
 
-        // Act
-        ToastManager.dismissAll();
+    // -----------------------------------------------------------------------------------------
+    // #region dismissAll
+    // -----------------------------------------------------------------------------------------
 
-        // Assert
-        expect(dismissAllMethodSpy).toHaveBeenCalled();
+    describe("dismissAll", () => {
```

Comment:
Just one tiny nitpick @SaidShah: could you alphabetize these regions? If the functions in the class are not alphabetized, they should be moved around too.

Reply:
Thanks for the feedback. I have updated the regions if you want to take another look. 👍